internal/util/ssa/cache.go

```diff
@@ -49,16 +54,21 @@ type Cache interface {
 
 	// Has checks if the given key (still) exists in the Cache.
 	// Note: keys expire after the ttl.
-	Has(key string) bool
+	Has(key, groupKind string) bool
 }
 
 // NewCache creates a new cache.
-func NewCache() Cache {
+func NewCache(opts ...NewCacheOption) Cache {
```

Comment:
I think I would simplify this a bit and just take a `controllerName string` parameter (we have this in a few places in controller-runtime as well, including for metrics of a controller / work queue)

Reply:
The values for this parameter would be:
* machinepool
* machinedeployment
* topology/cluster

(i.e. the names of the "owning" controllers)